test/integration/apiserver/apply/apply_crd_test.go

```diff
@@ -720,3 +723,113 @@ spec:
 	}
 	verifyReplicas(t, result, 1)
 }
+
+// TestApplyCRDStructuralSchema tests that when a CRD has a structural schema in its validation field,
+// it will be used to construct the CR schema used by apply.
+func TestApplyCRDStatus(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	server, err := apiservertesting.StartTestServer(t, apiservertesting.NewDefaultTestServerOptions(), nil, framework.SharedEtcd())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.TearDownFn()
+	config := server.ClientConfig
+
+	apiExtensionClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	noxuDefinition := fixtures.NewMultipleVersionNoxuCRD(apiextensionsv1beta1.ClusterScoped)
+
+	var c apiextensionsv1beta1.CustomResourceValidation
+	err = json.Unmarshal([]byte(`{
```

Comment:
given we unmarshall the JSON blob directly and not re-use it, it seems better to populate the CustomResourceValidation directly and not unmarshall from JSON?

for reference: https://github.com/kubernetes/kubernetes/issues/81134
but this file holds test code.